.github/workflows/wheels-linux.yml

```diff
@@ -0,0 +1,55 @@
+# Build, test and publish wheels to PyPI
+# Build and test run on every push
+# Publish runs only on a tag push
+# Note: This does not publish sdist
+name: Wheels (Linux)
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    name: Build wheels on Linux
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        submodules: true
+
+    - uses: actions/setup-python@v2
+      name: Install Python
+      with:
+        python-version: 3.8
+
+    - name: Install cibuildwheel
+      run: |
+        python -m pip install cibuildwheel==1.4.2
+
+    - name: Build wheels
+      env:
+        # Skip building on Python 2.7 and PyPy
+        CIBW_SKIP: cp27-* pp*
+        CIBW_BEFORE_TEST: pip install -r tests/requirements.txt
+        CIBW_TEST_COMMAND: pytest {project}/tests
+      run: |
+        python -m cibuildwheel --output-dir dist
+
+    - uses: actions/upload-artifact@v2
```

Comment:
Hi where does this upload to?

Reply:
Github stores in temporary storage for the CI run. It is private to that particular run and gets cleaned later on. Specifically, in this run, we will later collect all these wheels in the publish step (which is only run when you push a git tag). If you setup PYPI_TOKEN in GitHub secrets, you can automatically publish to PyPI when pushing a new git tag.

Reply:
Good thanks for the clarification.